corporate/lib/stripe.py

```diff
@@ -5504,37 +5533,20 @@ def invoice_plans_as_needed(event_time: datetime | None = None) -> None:
             remote_server = plan.customer.remote_server
             billing_session = RemoteServerBillingSession(remote_server=remote_server)
 
-        assert plan.next_invoice_date is not None  # for mypy
-
         if plan.fixed_price is not None and not plan.reminder_to_review_plan_email_sent:
             maybe_send_fixed_price_plan_renewal_reminder_email(plan, billing_session)
 
+        try_to_invoice_plan = True
         if remote_server:
-            free_plan_with_no_next_plan = (
-                not plan.is_a_paid_plan() and plan.status == CustomerPlan.ACTIVE
```

Comment:
Switched this in the new helper function to use `is_complimentary_access_plan`, which didn't exist until a few months ago. But using the new function, it makes it clearer why we still process an invoice for this case.